fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2017,7 +2030,12 @@ private CompletableFuture<Boolean> checkVersion(@Nullable UserVersionChecker use
             storeHeaderFuture = subspacePreloadFuture.thenCombine(storeHeaderFuture, (vignore, storeHeader) -> storeHeader);
         }
         CompletableFuture<Boolean> result = storeHeaderFuture.thenCompose(storeHeader -> checkVersion(storeHeader, userVersionChecker));
-        return context.instrument(FDBStoreTimer.Events.CHECK_VERSION, result);
+        return context.instrument(FDBStoreTimer.Events.CHECK_VERSION, result).thenApply(versionChanged -> {
+            if (!this.versionChanged && versionChanged) {
+                this.versionChanged = true;
+            }
+            return versionChanged;
+        });
```

Comment:
I think this could be simplified to 
```suggestion
        return context.instrument(FDBStoreTimer.Events.CHECK_VERSION, result).thenApply(versionChanged -> {
            this.versionChanged |= versionChanged
            return versionChanged;
        });
```

But `|=` is rarely used, so being more explicit here, might improve readibility.

Reply:
done